src/api-service/__app__/onefuzzlib/secrets.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+
+from typing import Tuple, cast
+from urllib.parse import urlparse
+from uuid import uuid4
+
+from azure.keyvault.secrets import KeyVaultSecret
+from onefuzztypes.models import SecretAddress, SecretData
+from pydantic import BaseModel
+
+from .azure.creds import get_instance_name, get_keyvault_client
+
+
+def save_to_keyvault(secret_data: SecretData) -> None:
+    if isinstance(secret_data.secret, SecretAddress):
```

Comment:
This is very surprising to me. Why don't we save instances of `SecretAddress`?

Reply:
Is this because it's supposed to be idempotent?

Reply:
if the SecreSata instance already has a secretAddress it means that it was already saved to the keyvault. so no need to save it again